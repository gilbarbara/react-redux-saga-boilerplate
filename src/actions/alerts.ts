import { ReactNode } from 'react';
import { uuid } from '@gilbarbara/helpers';
import { createAction } from '@reduxjs/toolkit';

import { actionPayload } from '~/modules/helpers';

import { ActionTypes } from '~/literals';

import { ShowAlertOptions } from '~/types';

export const hideAlert = createAction<string>(ActionTypes.ALERTS_HIDE);

export const showAlert = createAction(
  ActionTypes.ALERTS_SHOW,
  (message: ReactNode, options: ShowAlertOptions) => {
    const timeout = options.type === 'error' ? 0 : 5;

    return actionPayload({
      id: options.id || uuid(),
      icon: options.icon || 'info-o',
      message,
      position: options.position || 'bottom-right',
      type: options.type || 'neutral',
      timeout: typeof options.timeout === 'number' ? options.timeout : timeout,
    });
  },
);
