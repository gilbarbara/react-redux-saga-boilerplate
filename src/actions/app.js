// @flow
/**
 * @module Actions/App
 * @desc App Actions
 */

import uuid from 'uuid/v4';
import { createActions } from 'redux-actions';

import { ActionTypes } from 'constants/index';

export { goBack, go, push, replace } from 'modules/history';

export const { hideAlert, showAlert, switchMenu } = createActions({
  [ActionTypes.SWITCH_MENU]: (query: string) => ({ query }),
  [ActionTypes.HIDE_ALERT]: (id: string) => ({ id }),
  [ActionTypes.SHOW_ALERT]: (message: string, options: Object) => {
    const timeout = options.variant === 'danger' ? 0 : 5;

    return {
      id: options.id || uuid(),
      icon: options.icon,
      message,
      position: options.position || 'bottom-right',
      variant: options.variant || 'dark',
      timeout: typeof options.timeout === 'number' ? options.timeout : timeout,
    };
  },
});
