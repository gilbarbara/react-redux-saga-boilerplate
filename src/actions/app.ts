/**
 * @module Actions/App
 * @desc App Actions
 */
import React from 'react';
import { nanoid } from 'nanoid';

import { createAction } from 'modules/helpers';

import { ActionTypes } from 'literals';

import { ShowAlertOptions } from 'types';

export { go, goBack, push, replace } from 'modules/history';

export const hideAlert = createAction(ActionTypes.HIDE_ALERT, (id: string) => ({ id }));

export const showAlert = createAction(
  ActionTypes.SHOW_ALERT,
  (message: React.ReactNode, options: ShowAlertOptions) => {
    const timeout = options.variant === 'danger' ? 0 : 5;

    return {
      id: options.id || nanoid(),
      icon: options.icon,
      message,
      position: options.position || 'bottom-right',
      variant: options.variant || 'dark',
      timeout: typeof options.timeout === 'number' ? options.timeout : timeout,
    };
  },
);
