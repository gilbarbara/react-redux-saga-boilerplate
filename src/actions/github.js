// @flow
/**
 * @module Actions/User
 * @desc User Actions
 */
import { createActions } from 'redux-actions';

import { ActionTypes } from 'constants/index';

export const { githubGetRepos: getRepos } = createActions({
  [ActionTypes.GITHUB_GET_REPOS]: (query: string) => ({ query }),
});
