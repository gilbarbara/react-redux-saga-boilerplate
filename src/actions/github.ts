/**
 * @module Actions/User
 * @desc User Actions
 */
import { createAction } from 'modules/helpers';

import { ActionTypes } from 'literals';

export const getRepos = createAction(ActionTypes.GITHUB_GET_REPOS_REQUEST, (query: string) => ({
  query,
}));
