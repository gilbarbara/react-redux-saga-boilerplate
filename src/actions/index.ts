export * from './app';
export * from './github';
export * from './user';
