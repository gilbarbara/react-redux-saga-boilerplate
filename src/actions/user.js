// @flow
/**
 * @module Actions/User
 * @desc User Actions
 */
import { createActions } from 'redux-actions';

import { ActionTypes } from 'constants/index';

export const { userLogin: login, userLogout: logOut } = createActions({
  [ActionTypes.USER_LOGIN]: () => ({}),
  [ActionTypes.USER_LOGOUT]: () => ({}),
});
