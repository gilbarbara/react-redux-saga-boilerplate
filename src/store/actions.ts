export { alertHide, alertShow } from './slices/alerts';
export { setAppOptions } from './slices/app';
export { getRepos, getReposFailure, getReposSuccess } from './slices/github';
export { login, loginSuccess, logOut, logOutSuccess } from './slices/user';
