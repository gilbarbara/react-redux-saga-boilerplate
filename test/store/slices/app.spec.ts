import { emptyAction } from 'test-utils';

import { setAppOptions } from '~/actions';

import reducer from '~/store/slices/app';

describe('slices/app', () => {
  describe('actions', () => {
    describe(`${setAppOptions.type}`, () => {
      it('should return an action', () => {
        expect(setAppOptions({ query: 'react' })).toMatchSnapshot();
      });
    });
  });

  describe('reducer', () => {
    let app = reducer(undefined, emptyAction);

    it('should return the initial state', () => {
      expect(reducer(app, emptyAction)).toMatchSnapshot();
    });

    it(`should handle ${setAppOptions.type}`, () => {
      app = reducer(app, setAppOptions({ query: 'test' }));
      expect(app).toMatchSnapshot();
    });
  });
});
