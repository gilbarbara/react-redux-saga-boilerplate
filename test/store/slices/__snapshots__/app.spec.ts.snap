// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`slices/app > actions > app/setAppOptions > should return an action 1`] = `
{
  "payload": {
    "query": "react",
  },
  "type": "app/setAppOptions",
}
`;

exports[`slices/app > reducer > should handle app/setAppOptions 1`] = `
{
  "query": "test",
}
`;

exports[`slices/app > reducer > should return the initial state 1`] = `
{
  "query": "react",
}
`;
